### PR TITLE
[SPARK-33248][SQL][FOLLOWUP] Update migration guide to make clear what behavior changed and make variable names and configuration name more clearer

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Affress comment
https://github.com/apache/spark/pull/30156#discussion_r514894596
https://github.com/apache/spark/pull/30156#discussion_r514894838

### Why are the changes needed?
Make change more readable

### Does this PR introduce _any_ user-facing change?
In Spark 3.1, when script transformation output's value size less then schema size in default-serde mode(script transformation with row format of `ROW FORMAT DELIMITED`), Spark will pad NUll value to supplement data. In Spark 3.0 or earlier, Spark will do nothing and throw `ArrayIndexOutOfBoundsException`. To restore the behavior before Spark 3.1, you can set `spark.sql.legacy.transformationNotPadNullToSupplementData.enabled` to `true`.

### How was this patch tested?
Existed UT

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
FYI @cloud-fan @HyukjinKwon

File: docs/sql-migration-guide.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ license: |
   
   - In Spark 3.1, the `schema_of_json` and `schema_of_csv` functions return the schema in the SQL format in which field names are quoted. In Spark 3.0, the function returns a catalog string without field quoting and in lower case. 
  
-  - In Spark 3.1, when `spark.sql.legacy.transformationPadNullWhenValueLessThenSchema` is true, Spark will pad NULL value when script transformation's output value size less then schema size in default-serde mode(script transformation with row format of `ROW FORMAT DELIMITED`). If false, Spark will keep original behavior to throw `ArrayIndexOutOfBoundsException`.
+  - In Spark 3.1, when script transformation output's value size less then schema size in default-serde mode(script transformation with row format of `ROW FORMAT DELIMITED`), Spark will pad NUll value to supplement data. In Spark 3.0 or earlier, Spark will do nothing and throw `ArrayIndexOutOfBoundsException`. To restore the behavior before Spark 3.1, you can set `spark.sql.legacy.transformationNotPadNullToSupplementData.enabled` to `true`.
```
Comment (cloud-fan) [inline docs/sql-migration-guide.md:55]: `value size less then` -> `value size is less than`

Comment (cloud-fan) [reply docs/sql-migration-guide.md:55]:
```
In Spark 3.0 or earlier, Spark will do nothing and throw `ArrayIndexOutOfBoundsException`
```

I don't think we need a migration guide if we make a failure case working...

Comment (cloud-fan) [reply docs/sql-migration-guide.md:55]: And it's weird to provide a config to fail

Comment (AngersZhuuuu) [reply docs/sql-migration-guide.md]: > `value size less then` -> `value size is less than`

Done

Comment (AngersZhuuuu) [reply docs/sql-migration-guide.md]: > ```
> In Spark 3.0 or earlier, Spark will do nothing and throw `ArrayIndexOutOfBoundsException`
> ```
> 
> I don't think we need a migration guide if we make a failure case working...

This is a long standing error, and it's reasonable to point out that this is a problem that users might inadvertently trigger when they use Script Transform but don't know why. Inconsistent behavior in reversion upgrades can be confusing, and it's nice to point out specifically

**Comment by HyukjinKwon** [comment]
@AngersZhuuuu, can you fix the PR title to reflect what your PR proposes? e.g.) fixing migration guide, changing the default, etc.

**Comment by AngersZhuuuu** [comment]
> @AngersZhuuuu, can you fix the PR title to reflect what your PR proposes? e.g.) fixing migration guide, changing the default, etc.

How about current.

**Comment by srowen** [comment]
This legacy config was never released in 3.0 right? because it is inverted here - seems OK either way though.

File: docs/sql-migration-guide.md [hunk 2]
```diff
@@ -52,7 +52,7 @@ license: |
   
   - In Spark 3.1, the `schema_of_json` and `schema_of_csv` functions return the schema in the SQL format in which field names are quoted. In Spark 3.0, the function returns a catalog string without field quoting and in lower case. 
  
-  - In Spark 3.1, when `spark.sql.legacy.transformationPadNullWhenValueLessThenSchema` is true, Spark will pad NULL value when script transformation's output value size less then schema size in default-serde mode(script transformation with row format of `ROW FORMAT DELIMITED`). If false, Spark will keep original behavior to throw `ArrayIndexOutOfBoundsException`.
+  - In Spark 3.1, when script transformation output's value size is less then schema size in default-serde mode(script transformation with row format of `ROW FORMAT DELIMITED`), Spark will pad NUll value to supplement data. In Spark 3.0 or earlier, Spark will do nothing and throw `ArrayIndexOutOfBoundsException`. To restore the behavior before Spark 3.1, you can set `spark.sql.legacy.transformationNotPadNullToSupplementData.enabled` to `true`.
```
Comment (cloud-fan) [inline docs/sql-migration-guide.md:55]: I do think throwing `ArrayIndexOutOfBoundsException` is a bug. Do we still throw it somewhere in the script transformation code path?

Comment (HyukjinKwon) [reply docs/sql-migration-guide.md:55]: I think it's arguable to call it a bug. That's why I suggested to add the migration guide here. The behaviour is just mismatch compared to Hive's behaviour. The previous behaviour was not padding null with the failfast.

Comment (cloud-fan) [reply docs/sql-migration-guide.md:55]: If we intend to fail, we should give a better error message with `AnalysisException`, not bug-like exceptions such as `ArrayIndexOutOfBoundsException`, NPE, etc.

Comment (HyukjinKwon) [reply docs/sql-migration-guide.md:55]: Yeah that might be the best for completeness. I am just elaborating why I think it should be best to update the migration guide. We'll move to this new behaviour by default and probably remove it away in the future anyway.

Comment (cloud-fan) [reply docs/sql-migration-guide.md:55]: Yea I'm just saying that we usually don't add migration guide if we make something failed to work, as it's not a breaking change and users have nothing to do for migration.

Comment (AngersZhuuuu) [reply docs/sql-migration-guide.md:55]: So, What should I do next? Personally, we need to let users know about this change

Comment (srowen) [reply docs/sql-migration-guide.md:55]: Nit: NUll -> NULL
Yeah, why would you ever set this flag to the original behavior, to fail? this is just a bug fix. Nobody can reasonably be relying on this behavior.

Comment (AngersZhuuuu) [reply docs/sql-migration-guide.md:55]: All  right, we  can just revert last pr.

**Pull request closed without merging**